common/display/displayqueue.cpp

```diff
@@ -322,4 +340,181 @@ bool DisplayQueue::CheckPlaneFormat(uint32_t format) {
+  result = pow(value, gamma);
+  if (result < 0.0)
+    result = 0.0;
+  if (result > 1.0)
+    result = 1.0;
+
+  return result;
+}
+
+void DisplayQueue::SetColorCorrection(struct gamma_colors gamma,
+                                      uint32_t contrast_c,
+                                      uint32_t brightness_c) const {
+  struct drm_color_lut* lut;
+  float brightness[3];
+  float contrast[3];
+  uint8_t temp[3];
+  int32_t ret;
+
+  /* reset lut when contrast and brightness are all 0 */
+  if (contrast_c == 0 && brightness_c == 0) {
+    lut = NULL;
+    ApplyPendingLUT(lut);
+    free(lut);
+    return;
+  }
+
+  lut = (struct drm_color_lut*)malloc(sizeof(struct drm_color_lut) * lut_size_);
+  if (!lut) {
+    ETRACE("Cannot allocate LUT memory");
+    return;
+  }
+
+  /* Unpack brightness values for each channel */
+  temp[0] = (brightness_c >> 16) & 0xFF;
+  temp[1] = (brightness_c >> 8) & 0xFF;
+  temp[2] = (brightness_c)&0xFF;
+
+  /* Map brightness from -128 - 127 range into -0.5 - 0.5 range */
+  brightness[0] = (float)(temp[0]) / 255 - 0.5;
+  brightness[1] = (float)(temp[1]) / 255 - 0.5;
+  brightness[2] = (float)(temp[2]) / 255 - 0.5;
+
+  /* Unpack contrast values for each channel */
+  temp[0] = (contrast_c >> 16) & 0xFF;
+  temp[1] = (contrast_c >> 8) & 0xFF;
+  temp[2] = (contrast_c)&0xFF;
+
+  /* Map contrast from 0 - 255 range into 0.0 - 2.0 range */
+  contrast[0] = (float)(temp[0]) / 128;
+  contrast[1] = (float)(temp[1]) / 128;
+  contrast[2] = (float)(temp[2]) / 128;
+
+  uint32_t max_value = (1 << 16) - 1;
+  uint32_t mask = ((1 << 8) - 1) << 8;
+  for (uint64_t i = 0; i < lut_size_; i++) {
+    /* Set lut[0] as 0 always as the darkest color should has brightness 0 */
+    if (i == 0) {
+      lut[i].red = 0;
+      lut[i].green = 0;
+      lut[i].blue = 0;
+      continue;
+    }
+
+    lut[i].red = 0xFFFF * TransformGamma(TransformContrastBrightness(
+                                             (float)(i) / lut_size_,
+                                             brightness[0], contrast[0]),
+                                         gamma.red);
+    lut[i].green = 0xFFFF * TransformGamma(TransformContrastBrightness(
+                                               (float)(i) / lut_size_,
+                                               brightness[1], contrast[1]),
+                                           gamma.green);
+    lut[i].blue = 0xFFFF * TransformGamma(TransformContrastBrightness(
+                                              (float)(i) / lut_size_,
+                                              brightness[2], contrast[2]),
+                                          gamma.blue);
+  }
+
+  ApplyPendingLUT(lut);
+  free(lut);
+}
+
+bool DisplayQueue::SetBroadcastRGB(const char* range_property) {
+  uint64_t p_value = -1;
+
+  drmModePropertyPtr p = drmModeGetProperty(gpu_fd_, broadcastrgb_id_);
+
+  if (!(p->flags & DRM_MODE_PROP_ENUM))
+    return false;
+
+  for (int i = 0; i < p->count_enums; i++) {
```

Comment:
@kalyankondapally ,range_property is a dynamic parameter, and SetBroadcastRGB may happens anytime instead of display initialize. so i need map range_property to p_value dynamically. is that your concerns?